src/components/CortexTeamActionItemsWidget/CortexTeamActionItemsWidget.tsx

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2022 Cortex Applications, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useAsyncEntity } from '@backstage/plugin-catalog-react';
+import { useCortexApi } from '../../utils/hooks';
+import { stringifyAnyEntityRef } from '../../utils/types';
+import {
+  EmptyState,
+  InfoCard,
+  Link,
+  Progress,
+  WarningPanel,
+} from '@backstage/core-components';
+import React from 'react';
+import { isUndefined } from 'lodash';
+import { MetadataItem } from '../MetadataItem';
+import { DefaultEntityRefLink } from '../DefaultEntityLink';
+import { useRouteRef } from '@backstage/core-plugin-api';
+import { initiativeRouteRef } from '../../routes';
+import { Grid, TableCell, TableRow } from '@material-ui/core';
+
+export const CortexTeamActionItemsWidget = () => {
+  const {
+    entity,
+    loading: entityLoading,
+    error: entityError,
+  } = useAsyncEntity();
+  const initiativeRef = useRouteRef(initiativeRouteRef);
+
+  const {
+    value: serviceInitiativeActionItems,
+    loading: actionItemsLoading,
+    error: actionItemsError,
+  } = useCortexApi(
+    async cortexApi => {
+      return entity !== undefined
+        ? await cortexApi.getInitiativeActionItemsForTeam(
+            stringifyAnyEntityRef(entity),
+          )
+        : undefined;
+    },
+    [entity],
+  );
+
+  if (entityLoading || actionItemsLoading) {
```

Comment:
we're starting to do this pattern literally everywhere, i feel like we should just have a HoC that takes a `f: (api: CortexApi) => Promise<T>` and a `component` (and error messages, etc) and does all this stuff internally

we can leave the `EmptyState` logic here, but at least the loading/errors can be factored out